### PR TITLE
Refactors code for Functor API

### PR DESCRIPTION
**Pull request by zygmuntszpak**
This is the basic template, and the rest should be relatively straightforward. 

I have commented out the `Aqua` related tests in `runtest` since I can't make sense of the results yet, and also they appear to have been preventing the execution of the HistogramThresholding tests.

### CODE REVIEW

File: src/deprecations.jl [hunk 1]
```diff
@@ -0,0 +1,6 @@
+using Base: depwarn
+
+function find_threshold(algorithm::Otsu,  histogram::AbstractArray, edges::AbstractRange)
+    depwarn("find_threshold(alg, histogram, edges) is deprecated, use find_threshold(alg, histogram, edges) instead", :find_threshold)
```
Comment (johnnychen94) [inline src/deprecations.jl:4]: 😆 

```suggestion
    depwarn("`find_threshold(alg, histogram, edges)` is deprecated, use `find_threshold(histogram, edges, alg)` instead", :find_threshold)
```

File: src/algorithms/otsu.jl [hunk 2]
```diff
@@ -1,6 +1,6 @@
 @doc raw"""
 ```
-t = find_threshold(Otsu(), histogram, edges)
+t = Otsu(histogram, edges)
 ```
```
Comment (johnnychen94) [inline src/algorithms/otsu.jl:4]: The signature is actually not `Otsu(histogram, edges)` but `(::Otsu)(histogram, edges)`. How about:

```suggestion
    t = find_threshold(histogram, edges, Otsu())
```

File: src/algorithms/otsu.jl [hunk 3]
```diff
@@ -11,6 +11,10 @@ set so that the resultant between-class variance is maximal.
 Returns a real number `t` in `edges`. The `edges` parameter represents an
 `AbstractRange` which specifies the intervals associated with the histogram bins.
 
+You can access further algorithm details via the extended help function by typing
+`??Otsu`
```
Comment (johnnychen94) [inline src/algorithms/otsu.jl:15]: This is automatically printed so no need to write it manually.

```
Extended help is available with `??`
```

```suggestion
```

File: Project.toml [hunk 4]
```diff
@@ -1,23 +1,26 @@
 name = "HistogramThresholding"
 uuid = "2c695a8d-9458-5d45-9878-1b8a99cf7853"
-authors = ["Dr. Zygmunt L. Szpak <zygmunt.szpak@gmail.com>"]
-version = "0.2.5"
+authors = ["Dr. Zygmunt L. Szpak <zygmunt.szpak@gmail.com> and contributors."]
```
Comment (johnnychen94) [inline Project.toml:3]: I think usually we just list the leading/original authors here.

The `authors` is actually a list so it should be multiple strings (the [TOML format](https://toml.io/en/)):

```suggestion
authors = ["Dr. Zygmunt L. Szpak <zygmunt.szpak@gmail.com>", "JuliaImages"]
```

for example, https://github.com/JuliaTesting/ReferenceTests.jl/blob/fd8949f4ba648aa31b8321163a54aa076d65bd82/Project.toml#L3

File: src/algorithms/otsu.jl (hunk 2 above)
Comment (zygmuntszpak) [reply src/algorithms/otsu.jl:4]: Ah yes, of course! I also forgot to update the example in that docstring.

File: src/algorithms/otsu.jl [hunk 5]
```diff
@@ -1,6 +1,6 @@
 @doc raw"""
 ```
-t = find_threshold(Otsu(), histogram, edges)
+t = find_threshold(histogram, edges, Otsu())
```
Comment (johnnychen94) [inline src/algorithms/otsu.jl:3]: Sorry I missed this. Maybe we should also mention the image version here:

```suggestion
t = find_threshold(histogram, edges, Otsu())
t = find_threshold(img, Otsu(); nbins=256)
```

**Pull request merged**